### PR TITLE
Create automatic documentation based on the cop's descriptions

### PR DESCRIPTION
**Pull request by MassimilianoLattanzio**
This PR adds the ability to create docs based on the cop's description directly.

To create new docs, run the `bundle exec rake` task, automatically creating all the relative files and checking if the provided examples contain any ruby parsing error.

This PR also contains some improvements, not strictly related to the PR topic:
- [Fix wrong changelog uri in gemspec](https://github.com/solidusio/rubocop-solidus/pull/51/commits/8d6bb9fa50f3013466ee9e43cee92e06fef894d4)
- [Remove rubocop extensions suggestions](https://github.com/solidusio/rubocop-solidus/pull/51/commits/9b51b5ce43dd56ba55014f7f477a0c4834c03822)
- [Improve CI GH action](https://github.com/solidusio/rubocop-solidus/pull/51/commits/3265db7699563b16d9efc08e79cd9511494c8f7d)
- [Improve README readability](https://github.com/solidusio/rubocop-solidus/pull/51/commits/767b90eb5dcbc735404c24c77b9e0fe7579d96ab)

### CODE REVIEW

**Comment by MassimilianoLattanzio** [comment]
> I am finding this PR hard to review -
> 
> * Seems like the PR is doing much more what the title of the PR states
> * In case, it's doing other stuff, can we mention it in the root comment of the PR as bullet points?
> * Some commit messages are not stating why the change is needed / have been added

I've updated the PR with your suggestion. Let me know if now everything is clear.

File: rubocop-solidus.gemspec [hunk 1]
```diff
@@ -20,6 +20,7 @@ Gem::Specification.new do |spec|
   spec.metadata['homepage_uri'] = spec.homepage
   spec.metadata['source_code_uri'] = 'https://www.github.com/solidusio/rubocop-solidus'
   spec.metadata['changelog_uri'] = 'https://www.github.com/solidusio/rubocop-solidus'
+  spec.metadata['documentation_uri'] = 'https://www.github.com/solidusio/rubocop-solidus/blob/master/docs/cops.md'
```
Comment (abhishekgupta5) [inline rubocop-solidus.gemspec:23]: nice

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -23,9 +23,11 @@ jobs:
...
-      - name: Install dependencies
-        run: |
-          gem install bundler
-          bundle install --jobs 4 --retry 3
-      - name: Run the default task
-        run: bundle exec rake
+      - name: Lint  files
+        run: bundle exec rake rubocop
+      - name: Run tests
+        run: bundle exec rake spec
```
Comment (abhishekgupta5) [inline .github/workflows/ci.yml:29]: Is this same as - `bundle exec rspec spec`?

**Review by abhishekgupta5 — APPROVED** [review]
Nice PR. LGTM

File: .github/workflows/ci.yml (hunk 2 above)
Comment (MassimilianoLattanzio) [reply .github/workflows/ci.yml:29]: Yes!

https://github.com/solidusio/rubocop-solidus/blob/c62eb78b1e5bbb77b46f2f91ed64c6e493cdb78f/Rakefile#L13-L15

**Pull request merged**